clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/StoreIngestionTask.java

```diff
@@ -1562,9 +1562,13 @@ private void processIngestionException() {
                     Collections.singletonList(HelixUtils.getPartitionName(kafkaVersionTopic, exceptionPartition)));
           } catch (HelixException helixException) {
             LOGGER.error(
-                "Got exception while marking replica status to ERROR for: {}",
+                "Got HelixException while trying to set replica status to ERROR for: {}",
                 getReplicaId(kafkaVersionTopic, exceptionPartition),
                 helixException);
+            // If the replica is not completed, report error to release the latch to unblock pending ST
+            if (!partitionConsumptionState.isCompletionReported()) {
```

Comment:
Yes the problem as discussed offline is still - we should account for partition that's completed, other wise it will just lagged but not reset.